clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11630,6 +11630,10 @@ def err_sycl_expected_finalize_method : Error<
 def ext_sycl_2020_attr_spelling : ExtWarn<
   "use of attribute %0 is a SYCL 2020 extension">,
   InGroup<Sycl2017Compat>;
+def err_sycl_taking_address_of_wrong_function : Error<
```

Comment:
@AlexeySachkov , WDYT?

Reply:
Personally I prefer the current message, because it is a bit more detailed. Maybe something like
`cannot take address of function %0 inside SYCL device code because it doesn't have 'intel::device_indirectly_callable' attribute`? It is similar to existing clang diagnostics for cases when address taking is not possible.

Reply:
>>Maybe something like
cannot take address of function %0 inside SYCL device code because it doesn't have 'intel::device_indirectly_callable' attribute?

I prefer the current message instead of this one.

Reply:
Existing message looks good to me as well as the second variant suggested by @Fznamznon.

Whichever variant we choose, I would like it:
- to mention SYCL device code explicitly, because user might be confused why one address of operation works fine, whilst other causes a compilation error
- to mention somehow that `intel::device_indirectly_callable` attribute should be applied to a function. Yes, user can read about it somewhere in documentation, but I would prefer compiler to be more verbose and user-friendly

I think that we can spell it similar to existing error about `SYCL_EXTERNAL` functions: `SYCL kernel cannot call an undefined function without SYCL_EXTERNAL attribute`. It is a bit flawed by itself, because beside SYCL kernel functions the same restriction applies to any device function, but the general structure looks good: `SYCL device kernel or device function cannot take an address of a function without 'intel::device_indirectly_callable'` attribute

Reply:
How about we leave it as it then?

Reply:
> How about we leave it "as is" then?

Works for me